evoHike.Backend/Controllers/TrailsController.cs

```diff
@@ -0,0 +1,18 @@
+﻿using evoHike.Backend.Data;
+using evoHike.Backend.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace evoHike.Backend.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class TrailsController(EvoHikeContext context) : ControllerBase
```

Comment:
A Controller-be nem jó dolog üzleti logikát, mint a DbContext lekérdezése rakni.

A Controller osztályokba én legfeljebb a bejövő requestnek az ellenőrzését, és hiba esetén különböző Httpresponse visszaadását raknám. Azonkívül viszont minden más csak egy hívás legyen a következő osztály (réteg) felé, ami majd lekérdezi az adatbázisból amit kell.